mbed-client/m2mbase.h

```diff
@@ -48,7 +49,7 @@ class M2MResource;
  *  This class is the base class based on which all LWM2M object models
  *  can be created. This serves base class for Object, ObjectInstances and Resources.
  */
-class M2MBase : public M2MReportObserver {
+class M2MBase : public M2MReportObserver, public M2MDynamicMemory {
```

Comment:
The namespace overload of new/delete seems to be prohibited by C++ standard. Anyways we could first make namespace changes and then try the overload in different environments. I have a separate ticket for namespace update. Maybe separate namespaces for application and core part. To me inheritance seems more clear way to do "dynamic memory options". It is a bit more brute force and maybe some complications with multiple inheritance might rise, but so far it has worked fine.